File: feign/src/main/java/dev/be/feign/controller/DemoController.java
package dev.be.feign.controller;

import dev.be.feign.service.DemoService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class DemoController {

    private final DemoService demoService;

    @GetMapping("/get")
    public String getController() {
        return demoService.get();
    }

    @PostMapping("/post")
    public String postController() {
        return demoService.post();
    }

    @GetMapping("/error")
    public String errorController() {
        return demoService.errorDecoder();
    }
}

File: module-api/src/main/kotlin/dev/be/moduleapi/controller/DemoController.kt
package dev.be.moduleapi.controller

import dev.be.moduleapi.service.DemoService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class DemoController(
    private val demoService: DemoService
) {

    @GetMapping("/save")
    fun save(): String {
        return demoService.save()
    }

    @GetMapping("/find")
    fun find(): String {
        return demoService.find()
    }

    @GetMapping("/exception")
    fun exception(): String {
        return demoService.exception()
    }

}